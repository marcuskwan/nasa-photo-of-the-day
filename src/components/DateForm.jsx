import React, { useState } from "react";
import PropTypes from "prop-types";

export default function DateForm({ date, setFetchDate }) {
  const [inputDate, setInputDate] = useState(date);
  console.log("inputDate:", inputDate);
  // handleChanges fn
  const handleChanges = event => setInputDate(event.target.value);
  // button that changes value's state to the chosen date
  const submitDate = event => {
    event.preventDefault();
    setFetchDate(inputDate)
  };
  if (!date) return <h3>Loading...</h3>;
  else
    return (
      <div className="date_form">
        <form>
          <input
            onChange={handleChanges}
            name="date_picker"
            type="date"
            value={inputDate}
          />
          <button onClick={submitDate}>Submit</button>
        </form>
      </div>
    );
}

DateForm.propTypes = {
  date: PropTypes.string,
  setInputDate: PropTypes.func,
};
