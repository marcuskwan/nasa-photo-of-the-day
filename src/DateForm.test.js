import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import DateForm from "./components/DateForm";

// it("renders without crashing", () => {
//   const div = document.createElement("div");
//   ReactDOM.render(<DateForm />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });
//snapshot code
describe("<DateForm />", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<DateForm />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
