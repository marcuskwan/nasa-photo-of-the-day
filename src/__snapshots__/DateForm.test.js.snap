// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="ui container"
>
  <div
    className="ui vertically divided middle aligned grid"
  >
    <div
      className="two column row"
    >
      <div
        className="column"
      >
        <img
          className="ui image"
          src="https://www.nasa.gov/sites/default/files/thumbnails/image/nasa-logo-web-rgb.png"
        />
      </div>
      <div
        className="center aligned column"
      >
        <div
          className="ui icon header"
        >
          <i
            aria-hidden="true"
            className="search icon"
            onClick={[Function]}
          />
          Search Date
        </div>
        <h3>
          Loading...
        </h3>
      </div>
    </div>
    <div
      className="one column row"
    >
      <div
        className="column"
      >
        <div
          className="title"
        >
          <h1 />
        </div>
        <div
          className="date"
        />
        <div
          className="explanation"
        />
        <div
          className="media_type"
        >
          media type: 
        </div>
        <div
          className="service_version"
        >
          service version: 
        </div>
        <div
          className="copyright"
        >
          copyright: 
        </div>
      </div>
    </div>
    <div
      className="one column row"
    >
      <div
        className="column"
      >
        <img
          className="ui image"
        />
      </div>
    </div>
  </div>
  <div>
    hi
  </div>
</div>
`;

exports[`<DateForm /> should match snapshot 1`] = `
<h3>
  Loading...
</h3>
`;
